dotenv-linter/src/schema.rs

```diff
@@ -0,0 +1,335 @@
+    }
+    #[test]
+    fn regex_good() {
+        let mut schema = load_schema();
+        schema.entries.get_mut("NAME").unwrap().regex = Some("^[ABCD]*$".to_string());
+        let lines: Vec<LineEntry> = vec![line_entry(1, 2, "NAME=BAD")];
+        let expected: Vec<Warning> = vec![];
+        let mut opt = CliOptions::default();
+        opt.schema = Some(schema);
+        assert_eq!(expected, crate::checks::run(&lines, &opt));
+    }
+    #[test]
+    fn regex_bad() {
+        let mut schema = load_schema();
+        schema.entries.get_mut("NAME").unwrap().regex = Some("^[ABCD]*$".to_string());
+        let lines: Vec<LineEntry> = vec![line_entry(1, 2, "NAME=joe")];
+        let expected: Vec<Warning> = vec![Warning::new(
+            1,
+            LintKind::SchemaViolation,
+            "The NAME key does not match the regex",
+        )];
+        let mut opt = CliOptions::default();
+        opt.schema = Some(schema);
+        assert_eq!(expected, crate::checks::run(&lines, &opt));
+    }
+    #[test]
+    fn boolean_good() {
+        let schema = load_schema();
+        let lines: Vec<LineEntry> = vec![line_entry(1, 2, "FLAG=true")];
+        let expected: Vec<Warning> = vec![];
+        let mut opt = CliOptions::default();
+        opt.schema = Some(schema);
+        assert_eq!(expected, crate::checks::run(&lines, &opt));
+    }
+    #[test]
+    fn boolean_bad() {
+        let schema = load_schema();
+        let lines: Vec<LineEntry> = vec![line_entry(1, 2, "FLAG=joe")];
+        let expected: Vec<Warning> = vec![Warning::new(
+            1,
+            LintKind::SchemaViolation,
+            "The FLAG key is not a valid boolean",
+        )];
+        let mut opt = CliOptions::default();
+        opt.schema = Some(schema);
+        assert_eq!(expected, crate::checks::run(&lines, &opt));
+    }
+    #[test]
+    fn create_file_schema() {
+        let json = r#"{
+            "version": "1.0.0",
+            "entries": [
+                {
+                    "key": "NAME",
+                    "type": "String"
+                },
+                {
+                    "key": "PORT",
+                    "type": "Number"
+                },
+                {
+                    "key": "URL",
+                    "type": "Url"
+                },
+                {
+                    "key": "EMAIL",
+                    "type": "Email"
+                },
+                {
+                    "key": "FLAG",
+                    "type": "Boolean"
+                }
+            ]
+        }"#;
+        // write the above json to a temp file
+        let temp_dir = tempdir().unwrap();
+        let file_path = temp_dir.path().join("schema.json");
+        let schema = {
+            let mut file = File::create(&file_path).unwrap();
+            file.write_all(json.as_bytes()).unwrap();
+            // load the schema from the file
+            DotEnvSchema::load(&file_path)
```

Comment:
I might suggest to use `tempfile` directly https://crates.io/crates/tempfile

Instead of tempdir here